### ISSUE TITLE
externs: CanvasRenderingContext2D.fillStyle should allow more types

### ISSUE THREAD

**Issue by myphysicslab** [google/closure-compiler#1998]
In the externs file `html5.js` the `fillStyle` property should allow the additional types `CanvasGradient` and `CanvasPattern`

```javascript
    /**
     * @type {string|CanvasGradient|CanvasPattern}
     * @implicitCast
     */
    CanvasRenderingContext2D.prototype.fillStyle;
```

See https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/fillStyle

Using NTI compiler I'm currently getting this error:

    src/lab/view/DisplayPolygon.js:405: ERROR - The right side in the assignment is not
    a subtype of the left side.
    Expected : string
    Found    : CanvasGradient|string
    More details:
    The found type is a union that includes an unexpected type: CanvasGradient
        context.fillStyle = this.fillStyle;
        ^

**Comment by fredj** [comment]
And the same applies to `CanvasRenderingContext2D.prototype.strokeStyle`

**Comment by Dominator008** [comment]
I think the underlying issue is https://github.com/google/closure-compiler/issues/1533, but I agree that the externs should be updated too.

**Comment by fredj** [comment]
fixed with #2003

**Comment by Dominator008** [comment]
Closed by #2003
